logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/Sequencer.java

```diff
@@ -85,7 +86,10 @@ public long tryWrite(final List<LogAppendEntry> appendEntries, final long source
     lock.lock();
     try {
       currentPosition = position;
-      isEnqueued = queue.offer(new SequencedBatch(currentPosition, sourcePosition, appendEntries));
+      isEnqueued =
+          queue.offer(
+              new SequencedBatch(
+                  ActorClock.currentTimeMillis(), currentPosition, sourcePosition, appendEntries));
```

Comment:
:upside_down_face:  I would make the "clock" injectable into the sequencer (with `ActorClock.currentTimeMillis` the default). This makes the whole thing more test-able. Soft suggestion.